src/Symfony/Component/MimePgp/PgpEncrypter.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\MimePgp;
+
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\MailboxListHeader;
+use Symfony\Component\Mime\Message;
+use Symfony\Component\MimePgp\Exception\KeyNotFoundException;
+use Symfony\Component\MimePgp\Mime\Part\Multipart\PgpEncryptedPart;
+use Symfony\Component\MimePgp\Mime\Part\PgpEncryptedInitializationPart;
+use Symfony\Component\MimePgp\Mime\Part\PgpEncryptedMessagePart;
+
+/*
+ * @author PuLLi <the@pulli.dev>
+ */
+final readonly class PgpEncrypter extends PgpMime
+{
+    /**
+     * @param array<string, string>                                                                              $recipientKeys
+     * @param array{binary?: string, cipher_algorithm?: string, digest_algorithm?: string, temp_prefix?: string} $options
+     */
+    public function __construct(private array $recipientKeys, array $options = [])
+    {
+        parent::__construct($options);
+    }
+
+    public function encrypt(Message $message): Message
+    {
+        $recipients = $this->getRecipients($message->getHeaders());
+        $missingPublicKeys = array_diff($recipients, array_keys($this->recipientKeys));
+        if (0 !== \count($missingPublicKeys)) {
+            throw new KeyNotFoundException(\sprintf('Public key not found for the following recipients: "%s"', implode(', ', $missingPublicKeys)));
+        }
+        $recipientKeys = array_intersect_key($this->recipientKeys, array_flip($recipients));
+        $body = $message->getBody();
+        $data = $this->iteratorToString($body->toIterable());
+
+        $output = $this->pgpProcess->encrypt($data, $recipientKeys);
```

Comment:
what happens when encrypting an email for multiple recipients ? Can we actually encrypt it with multiple keys (in a way requiring *one* of those keys to read it) ?

Reply:
With PGP/MIME, it is possible to send the same encrypted email to multiple recipients.